codec-http/src/test/java/io/netty5/handler/codec/http/HttpResponseDecoderTest.java

```diff
@@ -509,55 +509,58 @@ private static void testResponseWithContentLengthFragmented(byte[] header, int f
 
     @Test
     public void testWebSocketResponse() {
-        byte[] data = ("HTTP/1.1 101 WebSocket Protocol Handshake\r\n" +
-                "Upgrade: WebSocket\r\n" +
-                "Connection: Upgrade\r\n" +
-                "Sec-WebSocket-Origin: http://localhost:8080\r\n" +
-                "Sec-WebSocket-Location: ws://localhost/some/path\r\n" +
-                "\r\n" +
-                "1234567812345678").getBytes();
+        byte[] data = ("HTTP/1.1 101 Switching Protocols\r\n" +
+                       "Upgrade: websocket\r\n" +
+                       "Connection: Upgrade\r\n" +
+                       "Sec-WebSocket-Accept: s3pPLMBiTxaQ9kYGzzhZRbK+xOo= \r\n" +
+                       "Sec-WebSocket-Protocol: chat\r\n\r\n")
+                .getBytes(CharsetUtil.US_ASCII);
+
         EmbeddedChannel ch = new EmbeddedChannel(new HttpResponseDecoder());
-        ch.writeInbound(ch.bufferAllocator().allocate(data.length).writeBytes(data));
+        assertTrue(ch.writeInbound(ch.bufferAllocator().copyOf(data)));
 
         HttpResponse res = ch.readInbound();
         assertThat(res.protocolVersion(), sameInstance(HttpVersion.HTTP_1_1));
-        assertThat(res.status(), is(HttpResponseStatus.SWITCHING_PROTOCOLS));
-        try (HttpContent<?> content = ch.readInbound()) {
-            assertThat(content.payload().readableBytes(), is(16));
-        }
+        assertThat(res.status(), sameInstance(HttpResponseStatus.SWITCHING_PROTOCOLS));
+        assertEquals(4, res.headers().size());
+        assertTrue(res.headers().contains(HttpHeaderNames.UPGRADE, HttpHeaderValues.WEBSOCKET, true));
+        assertTrue(res.headers().contains(HttpHeaderNames.CONNECTION, HttpHeaderValues.UPGRADE, true));
+        assertTrue(res.headers().contains(HttpHeaderNames.SEC_WEBSOCKET_ACCEPT, "s3pPLMBiTxaQ9kYGzzhZRbK+xOo=",
+                                          false));
+        assertTrue(res.headers().contains(HttpHeaderNames.SEC_WEBSOCKET_PROTOCOL, "chat", false));
 
-        assertThat(ch.finish(), is(false));
+        LastHttpContent<?> lastHttpContent = ch.readInbound();
+        assertEquals(new EmptyLastHttpContent(ch.bufferAllocator()), lastHttpContent);
```

Comment:
I think the `lastHttpContent` and `new EmptyLastHttpContent()` should be closed as well.

Reply:
Done